ast/edit/option_editor_test.go

```diff
@@ -0,0 +1,242 @@
+					"foo":   values.New("foo"), // should ignore this
+					"bar":   values.New("bar"), // should ignore this
+					"buz":   values.New("buz"), // should ignore this
+					"every": values.New(duration("2h")),
+					"delay": values.New(duration("42m")),
+					"cron":  values.New("buz"),
+					"retry": values.NewInt(10),
+				}))
+			},
+		},
+		{
+			name:         "sets_option_to_array",
+			in:           `option foo = "edit me"`,
+			edited:       []string{`option foo = [1, 2, 3, 4]`},
+			editExpected: true,
+			edit: func(node ast.Node) (bool, error) {
+				literal := edit.GetLiteral(values.NewArrayWithBacking(semantic.Int, []values.Value{
+					values.NewInt(1),
+					values.NewInt(2),
+					values.NewInt(3),
+					values.NewInt(4),
+				}))
+				return edit.Edit(node, "foo", edit.EditValueFn(literal))
+			},
+		},
+		{
+			name: "sets_option_to_object",
+			in:   `option foo = "edit me"`,
+			edited: []string{
+				`option foo = {x: "x", y: "y"}`,
+				`option foo = {y: "y", x: "x"}`,
+			},
+			editExpected: true,
+			edit: func(node ast.Node) (bool, error) {
+				literal := edit.GetLiteral(values.NewObjectWithValues(map[string]values.Value{
+					"x": values.New("x"),
+					"y": values.New("y"),
+				}))
+				return edit.Edit(node, "foo", edit.EditValueFn(literal))
+			},
+		},
+		{
+			name: "sets_option_mixed",
+			in:   `option foo = "edit me"`,
+			edited: []string{
+				`option foo = {x: {a: [1, 2, 3]}, y: [[1], [2, 3]], z: [{a: 1}, {b: 2}]}`,
+				`option foo = {x: {a: [1, 2, 3]}, z: [{a: 1}, {b: 2}], y: [[1], [2, 3]]}`,
+				`option foo = {y: [[1], [2, 3]], x: {a: [1, 2, 3]}, z: [{a: 1}, {b: 2}]}`,
+				`option foo = {z: [{a: 1}, {b: 2}], x: {a: [1, 2, 3]}, y: [[1], [2, 3]]}`,
+				`option foo = {y: [[1], [2, 3]], z: [{a: 1}, {b: 2}], x: {a: [1, 2, 3]}}`,
+				`option foo = {z: [{a: 1}, {b: 2}], y: [[1], [2, 3]], x: {a: [1, 2, 3]}}`,
+			},
+			editExpected: true,
+			edit: func(node ast.Node) (bool, error) {
+				x := values.NewObjectWithValues(map[string]values.Value{
+					"a": values.NewArrayWithBacking(semantic.Array, []values.Value{
+						values.NewInt(1),
+						values.NewInt(2),
+						values.NewInt(3),
+					}),
+				})
+				y := values.NewArrayWithBacking(semantic.Array, []values.Value{
+					values.NewArrayWithBacking(semantic.Int, []values.Value{
+						values.NewInt(1),
+					}),
+					values.NewArrayWithBacking(semantic.Int, []values.Value{
+						values.NewInt(2),
+						values.NewInt(3),
+					}),
+				})
+				z := values.NewArrayWithBacking(semantic.Object, []values.Value{
+					values.NewObjectWithValues(map[string]values.Value{
+						"a": values.NewInt(1),
+					}),
+					values.NewObjectWithValues(map[string]values.Value{
+						"b": values.NewInt(2),
+					}),
+				})
+
+				literal := edit.GetLiteral(values.NewObjectWithValues(map[string]values.Value{
+					"x": x,
+					"y": y,
+					"z": z,
+				}))
+				return edit.Edit(node, "foo", edit.EditValueFn(literal))
```

Comment:
Its not clear to me why this produces multiple edits. Can you explain that?

Reply:
It's because `x, y, z` can happen in any order because they are passed as a map. Would it be better if I execute the comparison directly on the ast and not on formatting?